mergify_engine/worker.py

```diff
@@ -235,18 +237,55 @@ async def run_engine(
         logger.debug("engine in thread end")
 
 
+class BucketSelector(typing.Protocol):
+    async def next_org_bucket(self) -> typing.Optional[OrgBucketNameType]:
+        ...
+
+
+@dataclasses.dataclass
+class DedicatedOrgBucketSelector:
+    redis_stream: utils.RedisStream
+    owner_id: github_types.GitHubAccountIdType
+
+    def _is_org_bucket_for_me(self, org_bucket: bytes) -> bool:
+        owner_id = int(org_bucket.split(b"~")[1])
+        return owner_id == self.owner_id
+
+    async def next_org_bucket(self) -> typing.Optional[OrgBucketNameType]:
+        # FIXME(sileht): we can use ZSCORE instead of ZRANGEBYSCORE when we
+        # have drop all login/name from bucket names.
+        # NOTE(sileht): Should we drop the 30s latency ? That may impact the rate limit...
+        now = time.time()
+        for org_bucket in await self.redis_stream.zrangebyscore(
+            "streams",
+            min=0,
+            max=now,
+        ):
+            if self._is_org_bucket_for_me(org_bucket):
+                statsd.increment(
+                    "engine.streams.selected",
+                    tags=[f"worker_id:dedicated-{self.owner_id}"],
```

Comment:
```suggestion
                    tags=[f"worker_id:dedicated-{self.owner_id}", "dedicated"],
```
otherwise it's going to be hard to select them in DD.